packages/suite/package.json

```diff
@@ -32,9 +32,15 @@
         "@trezor/suite-data": "*",
         "@trezor/suite-desktop-api": "*",
         "@trezor/suite-storage": "*",
+        "@trezor/type-utils": "*",
         "@trezor/urls": "*",
         "@trezor/utils": "*",
         "@trezor/validation": "*",
+        "@suite-common/intl-types": "*",
+        "@suite-common/metadata-types": "*",
```

Comment:
Why not one package for all types? And structure them inside?

Reply:
Big packages usually leads to cyclic dependencies. Imagine this pattern:

1. I have `packageA` which have type `FormInput` and there are multiple forms in this package that needs this type
2. I have `packageB` which also have form that needs to use `FormInput` so you import it from `packageA`
3. Now you want to add this form, alongside others into to your main `packageA` but you can't because it will cause cyclic dependecy.

Now you have two options how to solve it:
1. You can merge `packageB` into `packageA`, but it will only amplify this cyclic deps issue for other packages. More things you will have in `packageA`, then more often you need to use `packageA` in other packages, but that will prevent you from importing any of that packages back into `packageA` because of cyclic dependency. That will force you to place everything into `packageA` which will grow into monolith (that's the exact thing happened in packages/suite).
2. You can create `packageC` which will contain this `FormInput` and both `packageA` and `packageB` can use it.

So creating smaller packages from start it's always better, because you have much lower chances to run into issue with cyclic dependencies, but not only that. Smaller packages gives you better control of what you will use in other packages, you can run smaller subsets test, lints etc which is faster. You can read more here https://www.notion.so/satoshilabs/Code-improvements-4-Packages-vs-folders-56fec5d5f02640b39dc2e314c255e465

From my experience I am pretty sure that in this specific case of `wallet-type` we will later run into these exact cyclic deps issues. Also making package just for `wallet-types` makes sense because, why would you want to have some absolutely unrelated intl types alongside types for wallet? Same goes for metadata vs wallet, they are not so different but still it's different enough to give them their own packages + again it will help in future prevent cyclic deps + all other package benefits mentioned before.

Reply:
But here we are speaking about the `-types` packages, so I don't fully understand how it would create more cyclic dependancies. 😞 The types package should not contain types derived from code in other packages, everything should be defined separately and then used in all sorts of places, always exported. Not sure how there could be more cyclic dependancies with one big `@suite-common/types` package, only less; when a type from one domain for some reason needs access to a type from another and vice versa.

Reply:
That's sadly not true nearly anywhere in suite, lot of types are derived from some constants or config, even this wallet and intl types.

There is only two ways how to solve it, create one giant package with everything (which we really don't want because we will end with another suite like folder) or create lot of smaller packages.

Another important thing is that we primary want to do per domain packages, not per type (reducers, constants, config, types...) as I explained in [other comment here](https://github.com/trezor/trezor-suite/pull/5799#discussion_r930341675) and splitting wallet domain into types just trick how to avoid giant wallet package.

Reply:
If this is the case, I would not even move them to a separate package before refactoring, honestly, and would much rather keep importing from the `suite` package than having a `-types` package depend on some code in `suite`...

Reply:
It doesn't depends on any code in suite now, all constants and config for `wallet-types` was also derived from suite to new packages here in common. I am getting little bit of lost in some many possibilities we discussed, so I will try to sum up:

1. I can put everything (types, constants, configs) into one `wallet` folder for sure, but I will need to add things later like utils, and `wallet-utils` contains like really lot of functions and that will again cause package to be much bigger that we want. And it works like gravity, later instead of creating new packages it will force me to put it directly here and it will grow, grow and grow.
2. I create one big package for all types. That is definitely not a good idea, some types will always be dependendant on something else and it could be really hard to separate them. In the end we would have end with one giant type package where will be lot of thing but also there will be low of exceptions placed somewhere else, because it was too hard to separate them, because they depended on something else. It would be super hard to maintain this and set up code rules about it with all these exceptions.
3. I can import types directly from `suite/`. But actually I can't. For example if I want to move transactions to common, I will need some types from `suite/`, but then again I need to use import and use that transations things back in `suite`. That' cyclic dependency, that's reason why we can't import nothing from suite.
4. I can do it as it is now. It's also not ideal too, because type package itself it quite big. I would prefer to have small per domain packages, but that something what would be very hard to refactor it to that shape now. 

From this list I would say only point 1 and 4 for are reasonable. Point 4 is actually very similar as point 1. but just with that extension that we are trying to keep smallest packages as possible, which in my experience is always better. Separating things from big package is problem as we can see now, but merging small packages together in future is really easy.